### PR TITLE
feat: extend i18n attributes to every content type

### PR DESCRIPTION
**Pull request by Marc-Roig**
### What does it do?

Extend every content type to have `locale` and `localizations` (the later will be completely removed during the v5 development) attributes.

### Why is it needed?

We will allow disabling i18n from a content type, but the locale attribute will always be present in the database. If i18n is disabled for the CT, the locale value will be `null`.

### How to test it?
See the response value of a content type with disabled i18n, it should have the locale attribute set to null. The i18n enabled content types should still keep a non null locale value.

### CODE REVIEW

File: packages/plugins/i18n/server/src/register.ts [hunk 1]
```diff
@@ -46,37 +46,35 @@ const addContentTypeSyncHooks = (strapi: Strapi) => {
...
-      _.set(attributes, 'localizations', {
-        writable: true,
-        private: false,
-        configurable: false,
-        visible: false,
-        type: 'relation',
-        relation: 'oneToMany',
-        target: contentType.uid,
-      });
+    _.set(attributes, 'localizations', {
```
Comment (alexandrebodin) [inline packages/plugins/i18n/server/src/register.ts:59]: Aren't we deleting those entirely ?

**Review by innerdvations — APPROVED** [review]
LGTM, we're removing 'localizations' in a later PR, correct?

File: packages/plugins/i18n/server/src/register.ts (hunk 1 above)
Comment (Marc-Roig) [reply packages/plugins/i18n/server/src/register.ts:59]: Yes we are going to be removing it in a later PR! 

I am keeping it for now to keep the UI stable and working (so we can select locales) It makes it easier for testing features atm!

**Comment by Marc-Roig** [comment]
@innerdvations yes!

**Review by jhoward1994 — APPROVED** [review]
Is it safe to remove the D&P condition at

https://github.com/strapi/strapi/blob/29255bc953ad5e2ca1458e157bdf55819b15696f/packages/core/content-releases/admin/src/components/CMReleasesContainer.tsx#L282

To fix the TS build in this PR?

**Comment by Marc-Roig** [comment]
@jhoward1994 lets see if this was fixed in v5/main , that error shouldnt come from this pr 🤞

**Pull request merged**